src/bt_peer_connection.cpp

```diff
@@ -664,7 +660,7 @@ namespace {
 		aux::write_uint32(crypto_field, write_buf);
 		aux::write_uint16(pad_size, write_buf); // len (pad)
 
-		std::generate(write_buf.data(), write_buf.data() + pad_size, random_byte);
+		aux::random_bytes({write_buf.data(), pad_size});
```

Comment:
those writes updates the span to refer to the remaining bytes. Perhaps it's not the most intuitive interface, but I think it's at least an improvement over the older versions that just increments a pointer/iterator. i.e. ``write_buf`` is passed by non-const reference there.